src/components/PopupWithForm.js

```diff
@@ -7,9 +7,17 @@ export default class PopupWithFrom extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form"); // selects modal form for specific instance of this obj
   }
 
+  setLoading(isLoading, submitSave) {
+    if (isLoading) {
+      this._popupElement.querySelector(".modal__save-button").textContent = "Saving...";
+    } else {
+      this._popupElement.querySelector(".modal__save-button").textContent = submitSave;
+    }
+  }
+
   close() {
-    this._popupForm.reset();
     super.close();
+    this._popupForm.reset();
     this._popupElement.removeEventListener("submit", this._handleFormSubmit);
   }
 
```

Comment:
`    this._inputList = document.querySelectorAll(".modal__input");
`

In line 26

You search in `document.querySelectorAll` but you need to find the inputs inside `this._popupForm` 


```js
    this._inputList = this._popupForm.querySelectorAll(".modal__input");


```